server/schedulers/evict_slow_store.go

```diff
@@ -0,0 +1,212 @@
+		conf := &evictSlowStoreSchedulerConfig{storage: storage, EvictedStores: make([]uint64, 0)}
+		if err := decoder(conf); err != nil {
+			return nil, err
+		}
+		return newEvictSlowStoreScheduler(opController, conf), nil
+	})
+}
+
+type evictSlowStoreSchedulerConfig struct {
+	storage       *core.Storage
+	EvictedStores []uint64 `json:"evict-stores"`
+}
+
+func (conf *evictSlowStoreSchedulerConfig) Persist() error {
+	name := conf.getSchedulerName()
+	data, err := schedule.EncodeConfig(conf)
+	failpoint.Inject("persistFail", func() {
+		err = errors.New("fail to persist")
+	})
+	if err != nil {
+		return err
+	}
+	return conf.storage.SaveScheduleConfig(name, data)
+}
+
+func (conf *evictSlowStoreSchedulerConfig) getSchedulerName() string {
+	return EvictSlowStoreName
+}
+
+type evictSlowStoreScheduler struct {
+	*BaseScheduler
+	conf *evictSlowStoreSchedulerConfig
+}
+
+func (s *evictSlowStoreScheduler) GetName() string {
+	return EvictSlowStoreName
+}
+
+func (s *evictSlowStoreScheduler) GetType() string {
+	return EvictSlowStoreType
+}
+
+func (s *evictSlowStoreScheduler) EncodeConfig() ([]byte, error) {
+	return schedule.EncodeConfig(s.conf)
+}
+
+func (s *evictSlowStoreScheduler) Prepare(cluster opt.Cluster) error {
+	if len(s.conf.EvictedStores) != 0 {
+		return s.prepareEvictLeader(cluster)
+	}
+	return nil
+}
+
+func (s *evictSlowStoreScheduler) Cleanup(cluster opt.Cluster) {
+	if len(s.conf.EvictedStores) != 0 {
+		s.cleanupEvictLeader(cluster)
+	}
+}
+
+func (s *evictSlowStoreScheduler) prepareEvictLeader(cluster opt.Cluster) error {
+	return cluster.SlowStoreEvicted(s.conf.EvictedStores[0])
+}
+
+func (s *evictSlowStoreScheduler) cleanupEvictLeader(cluster opt.Cluster) {
+	cluster.SlowStoreRecovered(s.conf.EvictedStores[0])
+}
+
+func (s *evictSlowStoreScheduler) schedulerEvictLeader(cluster opt.Cluster) []*operator.Operator {
+	storeMap := map[uint64][]core.KeyRange{
+		s.conf.EvictedStores[0]: {core.NewKeyRange("", "")},
+	}
+	return scheduleEvictLeaderBatch(s.GetName(), cluster, storeMap, EvictLeaderBatchSize)
+}
+
+func (s *evictSlowStoreScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
+	if len(s.conf.EvictedStores) != 0 {
+		allowed := s.OpController.OperatorCount(operator.OpLeader) < cluster.GetOpts().GetLeaderScheduleLimit()
+		if !allowed {
+			operator.OperatorLimitCounter.WithLabelValues(s.GetType(), operator.OpLeader.String()).Inc()
+		}
+		return allowed
+	}
+	return true
+}
+
+func (s *evictSlowStoreScheduler) Schedule(cluster opt.Cluster) []*operator.Operator {
+	schedulerCounter.WithLabelValues(s.GetName(), "schedule").Inc()
+	var ops []*operator.Operator
+
+	evictedStores := s.conf.EvictedStores
+	if len(evictedStores) != 0 {
+		store := cluster.GetStore(evictedStores[0])
+		if store == nil || store.IsTombstone() {
+			// Previous slow store had been removed, remove the sheduler and check
+			// slow node next time.
+			log.Info("slow store has been removed",
+				zap.Stringer("store", store.GetMeta()))
```

Comment:
Maybe id is enough? If you wanna print meta, I think you should use `ZapRedactStringer`.